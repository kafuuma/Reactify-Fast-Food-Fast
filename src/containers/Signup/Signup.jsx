import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import signupUser from '../../actions/authActions/signupActions';
import SignupForm from '../../components/signup/SignupForm';

export class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      full_name: '',
      email: '',
      contact: '',
      password: '',
      confirm_password: ''
    };
  }

  componentWillReceiveProps(nextProps) {
    const { message } = nextProps;
    if (message == 'signup successfull') {
      toast.success(message);
      const { history } = this.props;
      history.push('/login');
    } else {
      toast.error(message);
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { signupUser } = this.props;
    signupUser(this.state);
  };

  render() {
    return (
      <SignupForm
        handleChange={this.handleChange}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

Signup.propTypes = {
  signupUser: PropTypes.func
};

Signup.defaultProps = {
  signupUser: () => {}
};

export const mapStateToProps = state => {
  return {
    message: state.signup.user,
    time_flag: state.signup.time_flag
  };
};

export default connect(
  mapStateToProps,
  { signupUser }
)(Signup);
