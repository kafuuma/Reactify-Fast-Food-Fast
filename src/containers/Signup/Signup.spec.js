import React from 'react';
import { shallow } from 'enzyme';
import { Signup } from './Signup';

const state = {
  full_name: '',
  email: '',
  contact: '',
  password: '',
  confirm_password: ''
};

const toast = jest.fn;
const message = {
  message: 'login successful'
};

const props = {
  toast,
  message
};

describe('<Signup />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Signup />);
  });

  it('matches snapshort', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call handleChange ', () => {
    const instance = wrapper.instance();
    expect(
      instance.handleChange({
        preventDefault: jest.fn,
        target: {
          name: 'email'
        }
      })
    );
  });
  it('should call handleSubmit ', () => {
    const instance = wrapper.instance();
    expect(
      instance.handleSubmit({
        preventDefault: jest.fn
      })
    );
  });

  it('should recieve nect prosp ', () => {
    wrapper.setProps(props);
    expect(wrapper.state()).toEqual(state);
  });
});
