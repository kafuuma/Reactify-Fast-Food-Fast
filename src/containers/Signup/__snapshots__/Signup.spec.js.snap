// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> matches snapshort 1`] = `ShallowWrapper {}`;
